Stricter params to prevent triggering NoMethodErrors

martinemde opened the pull request.

martinemde commented on app/controllers/api/v1/hook_relay_controller.rb:
```diff
@@ -1,11 +1,10 @@
 class Api::V1::HookRelayController < Api::BaseController
-  before_action :set_hook_relay_report_params
```
I felt compelled to clean this controller up a bit. Sorry for the extra review overhead.

segiddins commented on app/controllers/concerns/params_require.rb:
```diff
@@ -0,0 +1,37 @@
+    raise ActionController::ParameterMissing, key, parameters.keys
+  end
+
+  # Safely require an Arrayl parameter without crashing on bad input.
```
```suggestion
  # Safely require an Array parameter without crashing on bad input.
```

martinemde replied:
I ended up removing this whole module, so this is fixed.

martinemde commented:
@segiddins I changed this pretty significantly. Re-requesting your review.

I threw out the module and replaced it with `params_fetch` that does everything we need and allowed me to clean up more than just the main ones.

I also made #4588, which uses params_fetch for everything. It's based on this PR. I like it, but I didn't want to block this PR on it because I think it might cross the line into non-obvious and confusing.
